src/Ledger/Utxo/Properties.lagda

```diff
@@ -71,42 +79,101 @@ newTxid⇒disj : txid tx ∉ map proj₁ (dom (utxo ˢ)) → disjoint' (dom (utx
+                     ≡ cbalance ((UTxOState.utxo utxoState) ∣ txins tx) + refunded utxoState tx
+consumedCoinEquality {tx} {utxoState} h = let utxo = UTxOState.utxo utxoState in begin
+  coin (balance (utxo ∣ txins tx) +ᵛ mint tx +ᵛ inject (refunded utxoState tx))
+    ≡⟨ ∙-homo-Coin coinIsMonoidMorphism _ _ ⟩
+  coin (balance (utxo ∣ txins tx) +ᵛ mint tx) + coin (inject $ refunded utxoState tx)
+    ≡⟨ cong (coin (balance (utxo ∣ txins tx) +ᵛ mint tx) +_) (property _) ⟩
+  coin (balance (utxo ∣ txins tx) +ᵛ mint tx) + refunded utxoState tx
+    ≡⟨ cong! (∙-homo-Coin coinIsMonoidMorphism _ _) ⟩
+  coin (balance (utxo ∣ txins tx)) + coin (mint tx) + refunded utxoState tx
+    ≡⟨ cong (λ x → cbalance (utxo ∣ txins tx) + x + refunded utxoState tx) h ⟩
+  cbalance (utxo ∣ txins tx) + 0 + refunded utxoState tx
+    ≡⟨ cong! (+-identityʳ (cbalance (utxo ∣ txins tx))) ⟩
+  cbalance (utxo ∣ txins tx) + refunded utxoState tx                        ∎
+
+producedCoinEquality : ∀ {pp} → coin (produced pp utxoState tx)
+                              ≡ cbalance (outs tx) + (txfee tx) + totalDeposits utxoState tx
+producedCoinEquality {utxoState} {tx} = begin
+  coin (balance (outs tx) +ᵛ inject (txfee tx) +ᵛ inject (totalDeposits utxoState tx))
+    ≡⟨ ∙-homo-Coin coinIsMonoidMorphism _ _ ⟩
+  coin (balance (outs tx) +ᵛ inject (txfee tx)) + coin (inject (totalDeposits utxoState tx))
+    ≡⟨ cong! (property _) ⟩
+  coin (balance (outs tx) +ᵛ inject (txfee tx)) + totalDeposits utxoState tx
+    ≡⟨ cong! (∙-homo-Coin coinIsMonoidMorphism _ _) ⟩
+  coin (balance (outs tx)) + coin (inject (txfee tx)) + totalDeposits utxoState tx
+    ≡⟨ cong (λ x → cbalance (outs tx) + x + totalDeposits utxoState tx) (property (txfee tx)) ⟩
+  cbalance (outs tx) + (txfee tx) + totalDeposits utxoState tx                     ∎
+
+balValueToCoin : ∀ {pp} → coin (mint tx) ≡ 0 → consumed pp utxoState tx ≡ produced pp utxoState tx
+               → cbalance ((UTxOState.utxo utxoState) ∣ txins tx) + refunded utxoState tx
+               ≡ cbalance (outs tx) + (txfee tx) + totalDeposits utxoState tx
+balValueToCoin {tx} {utxoState} {pp} h h' = begin
+  cbalance ((UTxOState.utxo utxoState) ∣ txins tx) + refunded utxoState tx
+    ≡˘⟨ consumedCoinEquality {tx} {utxoState} {pp} h ⟩
+  coin (consumed pp utxoState tx) ≡⟨ cong! h' ⟩
+  coin (produced pp utxoState tx) ≡⟨ producedCoinEquality {utxoState} {tx} {pp} ⟩
+  cbalance (outs tx) + (txfee tx) + totalDeposits utxoState tx ∎
+
+posPart-negPart≡x : ∀ {x} → posPart x ⊖ negPart x ≡ x
+posPart-negPart≡x {ℤ.+_ n}     = refl
+posPart-negPart≡x {ℤ.negsuc n} = refl
+
+module DepositHelpers
+  (utxo : UTxO) (fees : Coin) (deposits : (DepositPurpose × Credential) ↛ Coin) (tx : TxBody) where
+
+  private
+    dep = getCoin deposits
+    uDep = getCoin (updateDeposits (txcerts tx) deposits)
+    Δdep = depositsChange (txcerts tx) deposits
+    ref = refunded      ⟦ utxo , fees , deposits ⟧ᵘ tx
+    tot = totalDeposits ⟦ utxo , fees , deposits ⟧ᵘ tx
```

Comment:
I think `totalDeposits` is a bit of a confusing name, since it sounds like it's all of the deposits in the ledger state rather than the deposited amount in the transaction. Maybe `deposited` would be a better name?

Reply:
And then `deposits` can be called `allDeposits` or `stateDeposits` to make it less ambiguous.